src/prayer/AnswerPrayerForm/AnswerPrayerFormConnected.js

```diff
@@ -19,33 +36,70 @@ class AnswerPrayerFormConnected extends React.Component {
             currentUser: { profile: { photo = { uri: '' } } = {} } = {},
           } = {},
         }) => (
-          <Mutation mutation={ANSWER_PRAYER}>
-            {(answerPrayer) => (
-              <AnswerPrayerForm
-                loading={profileLoading}
-                onSubmit={(values) => {
-                  answerPrayer({
-                    variables: {
-                      id: values.id,
-                      answer: values.answer,
-                    },
-                    refetchQueries: () => [
-                      { query: GET_PRAYER_FEED, variables: { type: 'USER' } },
-                    ],
-                  });
-                  this.props.navigation.pop();
-                }}
-                avatarSource={photo}
-                {...this.props}
-                onClose={() => this.props.navigation.pop()}
-                title={"Celebrate God's faithfulness"}
-                prayerId={this.props.navigation.getParam('prayerId', '')}
-                prayerText={this.props.navigation.getParam('prayerText', '')}
-                prayerAnswer={this.props.navigation.getParam(
-                  'prayerAnswer',
-                  ''
+          <Mutation mutation={INTERACT_WITH_PRAYER}>
+            {(interactWithPrayer) => (
+              <Mutation mutation={ANSWER_PRAYER}>
+                {(answerPrayer) => (
+                  <AnswerPrayerForm
+                    loading={profileLoading}
+                    onSubmit={(values) => {
+                      answerPrayer({
+                        variables: {
+                          id: values.id,
+                          answer: values.answer,
+                        },
+                        refetchQueries: () => [
+                          {
+                            query: GET_PRAYER_FEED,
+                            variables: { type: 'USER' },
+                          },
+                        ],
+                      });
+                      this.props.navigation.pop();
+                    }}
+                    avatarSource={photo}
+                    {...this.props}
+                    onClose={() => this.props.navigation.pop()}
+                    title={"Celebrate God's faithfulness"}
+                    prayer={{
+                      id: this.prayerId,
+                      text: this.props.navigation.getParam('prayerText', ''),
+                      answer: this.prayerAnswer,
+                    }}
+                    action={
+                      this.prayerAnswer ? (
```

Comment:
why is this necessary? why would there not be an answer coming to the `AnswerPrayerForm`?

Reply:
oh duh. dumb question. nevermind.